physical/consul/consul.go

```diff
@@ -65,41 +30,23 @@ const (
 	consistencyModeStrong = "strong"
 )
 
-type notifyEvent struct{}
-
 // Verify ConsulBackend satisfies the correct interfaces
 var _ physical.Backend = (*ConsulBackend)(nil)
 var _ physical.HABackend = (*ConsulBackend)(nil)
 var _ physical.Lock = (*ConsulLock)(nil)
 var _ physical.Transactional = (*ConsulBackend)(nil)
-var _ physical.ServiceDiscovery = (*ConsulBackend)(nil)
-
-var (
-	hostnameRegex = regexp.MustCompile(`^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$`)
-)
+var _ sr.ServiceRegistration = (*ConsulBackend)(nil)
 
 // ConsulBackend is a physical backend that stores data at specific
 // prefix within Consul. It is used for most production situations as
 // it allows Vault to run on multiple machines in a highly-available manner.
 type ConsulBackend struct {
-	path                string
-	logger              log.Logger
-	client              *api.Client
```

Comment:
Should client be removed here? I suspect it's needed in this file.

Reply:
It has been removed, you are looking at the master branch here :-)

Reply:
I meant, i'm not sure it should be removed from consul.Backend